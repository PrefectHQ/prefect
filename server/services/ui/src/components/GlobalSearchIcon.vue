<script>
export default {
  props: {
    type: {
      required: true,
      type: String,
      validator: label => {
        return typeof label == 'string'
      }
    }
  },
  computed: {
    resultIcon() {
      switch (this.type) {
        case 'flow':
          return 'timeline'
        case 'flow_run':
          return 'trending_up'
        case 'task':
          return 'fiber_manual_record'
        case 'task_run':
          return 'done_all'
        default:
          return ''
      }
    }
  }
}
</script>

<template>
  <v-list-item-avatar>
    <v-icon>{{ resultIcon }}</v-icon>
  </v-list-item-avatar>
</template>
