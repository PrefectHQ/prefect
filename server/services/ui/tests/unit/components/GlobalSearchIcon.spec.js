// Libraries
import Vuetify from 'vuetify'
import Vue from 'vue'

// Components
import GlobalSearchIcon from '@/components/GlobalSearchIcon'

// Utilities
import { createLocalVue, shallowMount } from '@vue/test-utils'

const localVue = createLocalVue()
Vue.use(Vuetify)

describe('GlobalSearchIcon', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const wrapperProps = input => {
    return {
      localVue,
      vuetify,
      propsData: {
        type: input
      }
    }
  }

  let wrapper = shallowMount(GlobalSearchIcon, wrapperProps('flow'))

  it('renders a vue instance', () => {
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render the avatar element', () => {
    expect(wrapper.find('v-list-item-avatar-stub').exists()).toBe(true)
  })

  it('should render the icon element', () => {
    expect(wrapper.find('v-icon-stub').exists()).toBe(true)
  })

  it('should display the correct icon', () => {
    wrapper = shallowMount(GlobalSearchIcon, wrapperProps('flow'))
    expect(wrapper.find('v-icon-stub').text()).toBe('timeline')

    wrapper = shallowMount(GlobalSearchIcon, wrapperProps('flow_run'))
    expect(wrapper.find('v-icon-stub').text()).toBe('trending_up')

    wrapper = shallowMount(GlobalSearchIcon, wrapperProps('task'))
    expect(wrapper.find('v-icon-stub').text()).toBe('fiber_manual_record')

    wrapper = shallowMount(GlobalSearchIcon, wrapperProps('task_run'))
    expect(wrapper.find('v-icon-stub').text()).toBe('done_all')
  })
})
