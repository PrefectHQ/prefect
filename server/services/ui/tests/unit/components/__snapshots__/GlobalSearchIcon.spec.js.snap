// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalSearchIcon should match snapshot 1`] = `
<v-list-item-avatar-stub
  size="40"
>
  <v-icon-stub>
    timeline
  </v-icon-stub>
</v-list-item-avatar-stub>
`;
